elastic/apm-server
# Remove index management, always use data streams

## Motivation/summary

- Remove `apm-server.ilm.*` config
- Remove `apm-server.data_streams.enabled` config
- Remove `setup.*` config
- Remove `output.elasticsearch.{index,indices}` config
- Stop building, including, and using libbeat fields

## Checklist

~- [ ] Documentation has been updated~ Doc updates being handled separately

## How to test these changes

1. Run apm-server without any special config, other than Elasticsearch host & credentials
2. Install the APM integration package via Kibana
3. Send events to apm-server
4. Check events are indexed into data streams

## Related issues

Closes #5961
